snuba/datasets/entities/assign_reason.py

```diff
@@ -117,3 +151,25 @@ def check_result_out_of_order(
             return "RESULT_OUT_OF_ORDER"
 
     return None
+
+
+def check_nondeterministic_query(
+    data: Sequence[Mapping[str, Any]],
+    expected_data: Sequence[Mapping[str, Any]],
+    sort_fields: Sequence[str],
+) -> Optional[str]:
+    """
+    Returns NONDETERMINISTIC_QUERY if different rows are returned due
+    to a missing or non unique order by clause.
+    """
+    if len(data) == len(expected_data):
+        for idx in range(len(data)):
+            if data[idx] != expected_data[idx]:
+                if all(
```

Comment:
I'm not sure I understand your logic here. If you have these results:

```
[{"group_id": 1, "count": 1}, {"group_id": 2, "count": 2}] # actual
[{"group_id": 2, "count": 2}, {"group_id": 1, "count": 1}] # expected
```

This won't be listed as non deterministic because the "count" value is never equal at the same index. The test you have below seems to work just because "count" is always 1.

Reply:
It's intentional that this case is not classified as non deterministic. In your example, count is the sort field hence one of the results is clearly correct and one clearly wrong. So we don't put it into the non deterministic bucket because there would be another problem with the query if we were ever to get this result.

More generally, we only assign as non deterministic if the sort key values are exactly the same but something else is different, as this indicates that there is no uniquely correct result.